### PR TITLE
docs: add sp compilation and dependencies guide

### PR DESCRIPTION
**Pull request by sysvm**
### Description

add sp deployment guide

### Rationale

N/A

### CODE REVIEW

**Review by Fynnss — APPROVED** [review]

File: src/guide/storage-provider/run-book/run-testnet-SP-node.md [hunk 1]
```diff
@@ -103,14 +110,6 @@ FundingPrivateKey = ""
 SealPrivateKey = ""
 ApprovalPrivateKey = ""
 # block syncer configuration
-# signer configuration
-[SignerCfg]
-WhitelistCIDR = ["0.0.0.0/0"]
-GasLimit = 210000
-OperatorPrivateKey = "${SP_Operator_PrivKey}"
-FundingPrivateKey = "${SP_Funding_PrivKey}"
```
Comment (Fynnss) [inline src/guide/storage-provider/run-book/run-testnet-SP-node.md:111]: Please briefly describe the users of these accounts and the tokens they need to have.

For the funding account, it should have at least 1000BNB, for operator and seal account, it should have enough BNB to charge the gas fee. Make sure 10BNB at least. and for Approval account, there is no need to have BNB tokens.

Comment (sysvm) [reply src/guide/storage-provider/run-book/run-testnet-SP-node.md:111]: ok

File: src/guide/storage-provider/run-book/compile-dependences.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+### Object Storage
+
+Greenfield is a decentralized data storage system which uses object storage as the main data storage system.SP encapsulates object storage as [Piece Store](../services/piece-store.md) which provides common interfaces to be compatible with multiple object storage. Therefore, if a user want to join SP or test the function of SP, you must use an object storage system.
+
+The following lists the supported object storage systems:
+
+1. [AWS S3](https://aws.amazon.com/s3/): can be used in production environment.
+2. [MinIO](https://min.io/): can be used in production environment which is compatible with AWS S3.
+3. `Disk File`: this is used for experiencing the basic features of SP and understanding how SP works. The piece data created by SP cannot be getted within the network and can only be used on a single machine.
```
Comment (will-2012) [inline src/guide/storage-provider/run-book/compile-dependences.md:74]: Rename `Disk File` to `Posix FileSystem`?

**Review by will-2012 — COMMENTED** [review]
LGTM.

File: src/guide/storage-provider/run-book/compile-dependences.md (hunk 2 above)
Comment (sysvm) [reply src/guide/storage-provider/run-book/compile-dependences.md]: fixed

File: src/guide/storage-provider/run-book/compile-dependences.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+Commit  : 6eb30c3bda1a29fc97a4345559944c35cd560517
+Build   : go1.18.4 darwin amd64 2023-03-04 23:54
+
+# show the gnfd-sp help
+./gnfd-sp -h
+```
+
+## SP Dependencies
+
+If a user wants to start SP in local or testnet, you must prepare `RDBMS` and `Object Storage` to store piece data.
```
Comment (RenRick) [inline src/guide/storage-provider/run-book/compile-dependences.md:53]: Add "SPDB" and "BSDB" here.

**Review by RenRick — APPROVED** [review]

File: src/guide/storage-provider/run-book/compile-dependences.md (hunk 3 above)
Comment (sysvm) [reply src/guide/storage-provider/run-book/compile-dependences.md]: fixed

**Pull request merged**